### PR TITLE
ditch negative zero in ftest output

### LINKED ISSUE #461: `ftest` and `-0.0`
Currently, `ftest` uses the unadjusted R2, which is constrained by definition to be on [0, 1]. As it stands in the tests and doctests, we have several cases where the floating point results for a null model give an R2 of `-0.0`. That's fine numerically, but realistically we know that this should be positive zero. Moreover, if you use a different BLAS (e.g. MKL), then you do get nonnegative `0.0`. This means that the doctests and tests for `show` methods can break based on the BLAS in a way that's much more annoying to fix than simply using `isapprox` instead of `isequal`.

Should we replace 

https://github.com/JuliaStats/GLM.jl/blob/3f9c8e44f85d1d997c38a30580ac447ac9bac55a/src/ftest.jl#L162

with

```julia
    return FTestResult(Int(nobs(mods[1])), SSR, df, max.(r2.(mods), 0), fstat, pval)
```
?

This is in some sense an upstream issue (i.e. the definition of `StatsBase.r2`), but I'm hesitant to apply a constraint at that point because a clearly negative R2 could be a decent test for pathological method definitions of `deviance` and `nulldeviance` for a particular type.

### PR DESCRIPTION
**Pull request by palday**
closes #461.

I also updated some docstring tests so that they would pass. (cf. also #469)

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks. Just another though: maybe we should apply rounding only in `show`? The advantage is that we know we use 4 digits, so we can `-0.0000` to `0.0000` without applying an arbitrary threshold.

This wouldn't fix tests that check internal fields directly, but anyway if we only clamp small negative values to zero, we have to use `isapprox` with some tolerance to handle small positive values.

**Comment by palday** [comment]
Can do, but mathematically this is non-adjusted R2, so should be strictly non-negative. Negative zero is just a floating point issue.

File: src/ftest.jl [hunk 1]
```diff
@@ -187,15 +188,21 @@ function show(io::IO, ftr::FTestResult{N}) where N
     outrows[1, :] = ["", "DOF", "ΔDOF", "SSR", "ΔSSR",
                      "R²", "ΔR²", "F*", "p(>F)"]
 
+    # get rid of negative zero -- doesn't matter mathematically,
+    # but messes up doctests and various other things
+    # cf. Issue #461 
+
+    r2vals = [replace(@sprintf("%.4f", val),  "-0.0000" => "0.0000") for val in ftr.r2]
```
Comment (nalimilan) [inline src/ftest.jl:195]:
```suggestion
    r2vals = [replace(@sprintf("%.4f", val), "-0.0000" => "0.0000") for val in ftr.r2]
```

**Comment by nalimilan** [comment]
Yes, I agree, but the choice of an arbitrary threshold is annoying, and it seems safer to just store whatever `r2` returns to avoid hiding potential bugs in implementations.

**Review by nalimilan — APPROVED** [review]

**Pull request merged**